ModelUtil error in JAXBContext.newInstance()

There are jakarta.xml.bind:jakarta.xml.bind-api:jar:3.0.1 and org.glassfish.jaxb:jaxb-core:jar:3.0.2 in the flat classpath. 
When the javax.xml.namespace.QName is passed to JAXBContext.newInstance() to create JAXBContenxt, it throws the follow error :  

Caused by: jakarta.xml.bind.JAXBException: Package javax.xml.namespace with Jakarta XML Binding class javax.xml.namespace.QName defined in a module java.xml must be open to at least jakarta.xml.bind module.
	at jakarta.xml.bind.ModuleUtil.delegateAddOpensToImplModule(ModuleUtil.java:132)
	at jakarta.xml.bind.ContextFinder.newInstance(ContextFinder.java:270)
	at jakarta.xml.bind.ContextFinder.newInstance(ContextFinder.java:263)
	at jakarta.xml.bind.ContextFinder.find(ContextFinder.java:395)
	at jakarta.xml.bind.JAXBContext.newInstance(JAXBContext.java:691)
	at org.apache.cxf.common.jaxb.JAXBContextCache$2.run(JAXBContextCache.java:339)
	at org.apache.cxf.common.jaxb.JAXBContextCache$2.run(JAXBContextCache.java:337)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:569)
	at org.apache.cxf.common.jaxb.JAXBContextCache.createContext(JAXBContextCache.java:337)
	at org.apache.cxf.common.jaxb.JAXBContextCache.getCachedContextAndSchemas(JAXBContextCache.java:245)
	at org.apache.cxf.jaxb.JAXBDataBinding.createJAXBContextAndSchemas(JAXBDataBinding.java:499)
	at org.apache.cxf.jaxb.JAXBDataBinding.initialize(JAXBDataBinding.java:358)`

It looks the ModelUtil has issues to check if there is other module has the implementation [in these lines](https://github.com/eclipse-ee4j/jaxb-api/blob/3.0.1/jaxb-api/src/main/mr-jar/jakarta/xml/bind/ModuleUtil.java#L115-L119) : 
Should it return when impModule` and `jaxbModule` is the same here ? Or even skip this `ModelUtils` operations from `ContextFinder` when it says it's flat classpath ?

--- a/api/src/main/java/jakarta/xml/bind/ContextFinder.java
+++ b/api/src/main/java/jakarta/xml/bind/ContextFinder.java
@@ -302,16 +302,18 @@ class ContextFinder {
         String factoryName = classNameFromSystemProperties();
         if (factoryName != null) return newInstance(contextPath, contextPathClasses, factoryName, classLoader, properties);
 
-        Object factory = properties.get(factoryId);
-        if (factory != null) {
-            if (factory instanceof String)  {
-                factoryName = (String) factory;
-            } else {
-                throw new JAXBException(Messages.format(Messages.ILLEGAL_CAST, factory.getClass().getName(), "String"));
+        if (properties != null) {
+            Object factory = properties.get(factoryId);
+            if (factory != null) {
+                if (factory instanceof String) {
+                    factoryName = (String) factory;
+                } else {
+                    throw new JAXBException(Messages.format(Messages.ILLEGAL_CAST, factory.getClass().getName(), "String"));
+                }
             }
-        }
-        if (factoryName != null) {
-            return newInstance(contextPath, contextPathClasses, factoryName, classLoader, properties);
+            if (factoryName != null) {
+                return newInstance(contextPath, contextPathClasses, factoryName, classLoader, properties);
+            }
         }
 
         JAXBContextFactory obj = ServiceLoaderUtil.firstByServiceLoader(
@@ -344,16 +346,18 @@ class ContextFinder {
         String factoryClassName = classNameFromSystemProperties();
         if (factoryClassName != null) return newInstance(classes, properties, factoryClassName);
 
-        Object ctxFactory = properties.get(JAXBContext.JAXB_CONTEXT_FACTORY);
-        if (ctxFactory != null) {
-            if (ctxFactory instanceof String)  {
-                factoryClassName = (String) ctxFactory;
-            } else {
-                throw new JAXBException(Messages.format(Messages.ILLEGAL_CAST, ctxFactory.getClass().getName(), "String"));
+        if (properties != null) {
+            Object ctxFactory = properties.get(JAXBContext.JAXB_CONTEXT_FACTORY);
+            if (ctxFactory != null) {
+                if (ctxFactory instanceof String) {
+                    factoryClassName = (String) ctxFactory;
+                } else {
+                    throw new JAXBException(Messages.format(Messages.ILLEGAL_CAST, ctxFactory.getClass().getName(), "String"));
+                }
             }
-        }
-        if (factoryClassName != null) {
-            return newInstance(classes, properties, factoryClassName);
+            if (factoryClassName != null) {
+                return newInstance(classes, properties, factoryClassName);
+            }
         }
 
         JAXBContextFactory factory =

--- a/api/src/main/java/jakarta/xml/bind/ModuleUtil.java
+++ b/api/src/main/java/jakarta/xml/bind/ModuleUtil.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2017, 2021 Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2017, 2022 Oracle and/or its affiliates. All rights reserved.
  *
  * This program and the accompanying materials are made available under the
  * terms of the Eclipse Distribution License v. 1.0, which is available at
@@ -111,6 +111,14 @@ class ModuleUtil {
         final Module implModule = factorySPI.getModule();
 
         Module jaxbModule = JAXBContext.class.getModule();
+
+        if (!jaxbModule.isNamed()) {
+            //we are not on the module path, so assume class-path mode
+            if (LOGGER.isLoggable(Level.FINE)) {
+                LOGGER.log(Level.FINE, "Using jakarta.xml.bind-api on the class path.");
+            }
+            return;
+        }
 
         for (Class<?> cls : classes) {
             Class<?> jaxbClass = cls.isArray() ?

